### ISSUE TITLE
PLR1716 (boolean-chained-comparison) does not detect attribute access expressions

### ISSUE THREAD

**Issue by Hawken94** [astral-sh/ruff#25041]
## Summary

`PLR1716` (`boolean-chained-comparison`) only triggers for simple `Name` nodes but fails to detect equivalent patterns using attribute access (`Attribute` nodes).

## Example

Given the following code:

```python
# Detected ✅ — simple variable
if 0 < a and a <= b:
    pass

# NOT detected ❌ — attribute access
if 0 < obj.value and obj.value <= limit:
    pass

# Also NOT detected ❌ — common real-world pattern
if obj.value > 0 and obj.value <= limit:
    pass

## Reproduction
$ echo 'a = 1
b = 2
if 0 < a and a <= b:
    pass' | ruff check --select PLR1716 -
# Output: PLR1716 detected ✅

$ echo 'class X:
    y = 1
x = X()
b = 2
if 0 < x.y and x.y <= b:
    pass' | ruff check --select PLR1716 -
# Output: All checks passed! (not detected ❌)

### Version

ruff 0.15.12

**Comment by ntBre** [comment]
I think this was probably an intentional decision in pylint to avoid issues with side effects. Although it should be rare, an attribute access could have a side-effectful `__getattr__`, for example. The rule could also apply to subscript accesses, but these similarly delegate to `__getitem__`. I also tested against pylint with your example in `try.py`, which has the same behavior:

```console
❯ uvx pylint try.py --disable=all --enable=chained-comparison
************* Module try
try.py:2:3: R1716: Simplify chained comparison between the operands (chained-comparison)
```

I'd be open to other thoughts, and we could try making this change to see an ecosystem report if nothing else, but I think this is  plausibly working as intended and consistent with upstream.
